### PR TITLE
Rg photogram industrial 2

### PR DESCRIPTION
**Pull request by rayguna**

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#
+#  fk_rails_...  (fan_id => users.id)
+#  fk_rails_...  (photo_id => photos.id)
+#
+# app/models/like.rb
+
+class Like < ApplicationRecord
+  belongs_to :fan, class_name: "User", counter_cache: true
+  belongs_to :photo, counter_cache: true
+end
```
Comment (hcontr4) [inline app/models/like.rb]: Perhaps add validation to make sure user can't like same picture more than once

**Review by hcontr4 — APPROVED** [review]

File: app/models/follow_request.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#
+class FollowRequest < ApplicationRecord
+  belongs_to :recipient, class_name: "User"
+  belongs_to :sender, class_name: "User"
+
+  enum status: { pending: "pending", rejected: "rejected", accepted: "accepted" } 
+
+  #the above automatically defines:
+  # scope :accepted, -> {where(status: "accepted")}
+  # scope :accepted, -> {where(status: "accepted")}
```
Comment (thierrychau) [inline app/models/follow_request.rb:30]: Delete commented code

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+#  fk_rails_...  (photo_id => photos.id)
+#
+# app/models/like.rb
+
+class Like < ApplicationRecord
+  belongs_to :fan, class_name: "User", counter_cache: true
+  belongs_to :photo, counter_cache: true
+
+  # app/models/like.rb
+  validates :fan_id, uniqueness: { scope: :photo_id, message: "has already liked this photo" }
```
Comment (thierrychau) [inline app/models/like.rb:28]: Great job!

**Review by thierrychau — CHANGES_REQUESTED** [review]
Please see minor comments below, great job!

**Review by thierrychau — APPROVED** [review]

File: app/models/like.rb (hunk 1 above)
Comment (rayguna) [reply app/models/like.rb]: I have made the update.

File: app/models/follow_request.rb (hunk 2 above)
Comment (rayguna) [reply app/models/follow_request.rb:30]: The commented code has been deleted.

**Pull request merged**